tests/unit_and_functional/test_stage_execution.py

```diff
@@ -149,3 +154,30 @@ def test_run_stage_failure_raises_exception_for_failed_setup(
 ):
     with raises(BodyworkStageFailure, match="KeyError"):
         run_stage("stage_5", project_repo_connection_string)
+
+
+def test_infer_executable_type_type():
+    assert _infer_executable_type("train_model.ipynb") == ExecutableType.JUPYTER_NB
+    assert _infer_executable_type("train_model.py") == ExecutableType.PY_MODULE
+
+    with raises(ValueError, match=r"cannot execute train_model.exe"):
+        _infer_executable_type("train_model.exe")
+
+
+def test_run_stage_with_jupyter_notebook(
+    setup_bodywork_test_project: Iterable[bool],
+    project_repo_connection_string: str,
+    bodywork_output_dir: Path,
+):
+    try:
+        run_stage("stage_jupyter", project_repo_connection_string)
+        assert True
+    except Exception:
+        assert False
```

Comment:
Do we need to print the exception out here so we can see the exception in the test runner (CircleCI)?